### DISCUSSION TITLE
Fluid API Issues and Pull Requests Megathread

### DISCUSSION THREAD

**Discussion by ghost** [FabricMC/fabric-api#1230] [FAQ]
#1136 - Fluid API v1 PR. 

> This module adds a new FabricFlowableFluid class and FabricFlowableFluidBlock class. The new fluid class lets the user register only one fluid instead of registering a still fluid and a flowing fluid the Vanilla way. It also lets the user set the maximum level of the fluid via an abstract method. Both classes provide some optimizations over Vanilla's counterpart classes and contain some new methods that can be overriden for more control.

#632 - Original Proposal

> Fluids

> Any still fluid would be allowed to be stored in a fluid container.

> Fluid layers are a representation of a fluid and its amount. A fluid’s amount cannot be negative.
> int getAmount()

> Gets the amount of fluid this fluid layer holds.
> Fluid getFluid()

> Gets the fluid this fluid layer holds.
> Providers

> The FluidProvider interface would allow management of fluid layers.
> FluidLayer getLayer()

> Should return the fluid layer in a provider.
> boolean insert(Fluid fluid, int amount)

> Inserts as much of an amount of fluid as possible into the provider. Should return the whether the fluid could be inserted.

> Example:

> if (!this.shouldAllowFluid(fluid)) return false;
> if (this.getLayer().getAmount() + amount > this.getMaximumAmount()) return false;

> this.getLayer().setAmount(this.layer.getAmount() + amount);
> return true;

> boolean remove(Fluid fluid, int amount)

> Depletes an amount of fluid from a provider. If there was not enough fluid in the provider, should return false.

#1166 - Discussion Issue before GH discussions was a thing.

> This is an issue for discussing the fluid transfer api. Welcome here is discussion on implementation, api, documentation and use cases for this api. This is also loosely related to an item transfer api and the framework of an energy api (fabric api will not provide one explicitly). Discussion here should be on topic. If discussion is disorderly or misconduct is found this issue will be closed and locked to collaborators.

>What will Fabric API not do:
Redefine cauldrons to be 4 bottles.
Game breaking changes
Design an api which completely prohibits alternative implementations from existing.

**Comment by sylv256** [comment]
ohno

**Reply by ghost** [reply]
Mate, stop randomly commenting shitpost on stuff. This is Github not fabricord.

**Comment by OroArmor** [comment]
#1136 is for in world fluids, and has nothing to do with storage or the transfer of fluids
